seshdash/views.py

```diff
@@ -615,3 +616,77 @@ def get_latest_bom_data(request):
     latest_bom_data.append({"item": "Recent Contact", "value": get_timesince(latest_bom.time)})
 
     return HttpResponse(json.dumps(latest_bom_data))
+
+
+@login_required
+def historical_data(request):
+    
+    # If ajax request
+    if request.method == 'POST': 
+        sort_value = request.POST.get('sort_value', '')
+        historical_data = get_historical_dict(column=sort_value)
+        return HttpResponse(json.dumps(historical_data))
+   
+    # On page load
+    else:
+        sites = get_objects_for_user(request.user, 'seshdash.view_Sesh_Site')
+        active_site = sites[0]
+        context_dict = {}
+        context_dict['sites'] = sites
+        context_dict['site_id'] = 0
+        context_dict['active_site'] = active_site
+        return render(request, 'seshdash/historical-data.html', context_dict);
+
+
+def get_alerts_for_year(site):
+    """ Returns a json object containing the number of alerts for a year range """
+ 
+    # NOTE: When using datetime object date_range does not include the last date
+    now = datetime.datetime.now()
+    year_before = now - relativedelta(years=1)
+    number_of_alerts = Sesh_Alert.objects.filter(date__range=[
+                                                             time_utils.get_date_dashed(year_before),\
+							     time_utils.get_date_dashed(now)], site=site ).count()
+
+
+    
+    return number_of_alerts;
+ 
+def get_avg_field_year(site, field):
+    """ Returns the average of a field for a year range in Daily Data Point"""
+    avg_field_yield = Daily_Data_Point.objects.filter(site=site)[:365].aggregate(Avg(field)).values()[0]
+    return avg_field_yield
+
+def get_historical_dict(column='daily_pv_yield'):
+    unit = Daily_Data_Point.UNITS_DICTIONARY[column]
+    print "The unit is: ",
+    print unit
+    sites = Sesh_Site.objects.all();
```

Comment:
Getting all sites is a problem. Imagine there are many users and some sites belong to them and some sites don't